src/metpy/calc/kinematics.py

```diff
@@ -567,7 +567,18 @@ def frontogenesis(potential_temperature, u, v, dx=None, dy=None, x_dim=-1, y_dim
 
     # Compute the angle (beta) between the wind field and the gradient of potential temperature
     psi = 0.5 * np.arctan2(shrd, strd)
-    beta = np.arcsin((-ddx_theta * np.cos(psi) - ddy_theta * np.sin(psi)) / mag_theta)
+    arg = (-ddx_theta * np.cos(psi) - ddy_theta * np.sin(psi)) / mag_theta
```

Comment:
One option would be something like this:
```suggestion
    arg = (-ddx_theta * np.cos(psi) - ddy_theta * np.sin(psi))
    index = mag_theta != 0
    arg[index] /= mag_theta[index]
```

While searching for the `where` argument to [`np.divide`](https://numpy.org/doc/stable/reference/generated/numpy.divide.html), I found [a StackOverflow answer](http://stackoverflow.com/questions/26248654/ddg#37977222) suggesting
```suggestion
    arg = np.divide(-ddx_theta * np.cos(psi) - ddy_theta * np.sin(psi)), mag_theta, out=np.zeros_like(mag_theta), where=mag_theta!=0)
```
which looks much cleaner

Reply:
I like this approach too.

Reply:
Also, by moving to the reformulation below, we can remove both `-` in the expression for `arg`, since it's only used inside a `**2`. This would save a few temporaries.

Reply:
Using `np.divide` makes sense to me, too! I'll make this change.